src/cli/commands/force/lightning/lwc/preview.ts

```diff
@@ -87,53 +89,102 @@ export default class Preview extends Setup {
         `$ sfdx force:lightning:lwc:preview -p Android -t LWCEmu2 -n HellowWordComponent`
     ];
 
+    // NOTE: The following properties are just place holders to help with typescript compile.
+    protected title: string = '';
+    protected fulfilledMessage: string = '';
+    protected unfulfilledMessage: string = '';
+
+    private serverPort: string = '';
+    private deviceName: string = '';
+    private componentName: string = '';
+    private targetApp: string = '';
+    private projectDir: string = '';
+    private configFilePath: string = '';
+    private appConfig:
+        | IOSAppPreviewConfig
+        | AndroidAppPreviewConfig
+        | undefined;
+
     public async run(): Promise<any> {
-        const platform = this.flags.platform;
-        this.logger.info(`Preview command invoked for ${platform}`);
-        return Promise.all<void, SetupTestResult>([
-            this.validateAdditionalInputs(),
-            super.run()
-        ])
+        const logger = this.logger;
+        logger.info(`Preview command invoked for ${this.flags.platform}`);
+
+        return this.validateAdditionalInputs() // first validate input parameters
```

Comment:
Moved away from `Promise.all` to instead validate the input parameters first followed by verifying the setup stage. This is b/c if any of the input params are invalid then there is no point to also run the setup stage.